rfcs/0001-forward-ref-by-default.md

```diff
@@ -0,0 +1,145 @@
+- Start Date: 2020-12-27
+- Target Major Version: 11.x
+- RFC PR: [preactjs/rfcs#6](https://github.com/preactjs/rfcs/pull/6)
+- Implementation PR: [preactjs/preact#2850](https://github.com/preactjs/preact/pull/2850)
+
+# Forward ref by default
+
+## Summary
+
+Currently Preact will give you a reference to the virtual dom-node when passing ref
+to a functional component. This seems kind off counter-intuitive since there's not much
+a user can do with a reference to a vnode. This rfc suggests forwarding the ref into props
+for functional components so it's easier to attach to a dom-node. Currently there is a way
+to achieve this in the `preact/compat` package under the name of `forwardRef`, this is a higher
+order function that allows you to achieve this behavior. Problem with this function is that
+now all these components are marked as side-effectful since we invoke a function by default.
+This can grow bundle-size in certain bundlers.
```

Comment:
Just in case you're not aware of this, the Babel React preset includes a plugin to add pure comments to these calls. Maybe something similar should be implemented for Preact.

[Example](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAbwGbQO4EMoBMBKBTJAXziSghDgCIo90BjGSgbgCgW8APSWOOiAOwDO8ACoALYPwDmcALwk0mXAQAUKgJRyAfHBYBIADxZgANy0AJPABsrEAwHpjZluqZA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react&prettier=true&targets=Node-13.1&version=7.12.12&externalPlugins=)

[Plugin](https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-react-pure-annotations/src/index.js)

Reply:
Good point! The pure comment is widely supported in bundlers. I think the main motivation behind this RFC is that component libraries tend to wrap every component with `forwardRef` (see Material UI). If every component needs to be wrapped, then there is something wrong and we should add that behavior by default.

Reply:
I think we should only list a quick summery of the changes here (1-2 sentences) and explain the reasoning in detail down below under the motivation headline.